src/modules/UI/scenes/Plugins/EdgeProvider.js

```diff
@@ -140,20 +141,24 @@ export class EdgeProvider extends Bridgeable {
   async getCurrentWalletInfo(): Promise<WalletDetails> {
     const edgeWallet = this._state.core.account.currencyWallets[this._state.ui.wallets.selectedWalletId]
     const currencyCode = this._state.ui.wallets.selectedCurrencyCode
-    let walletName = edgeWallet.name ? edgeWallet.name : ''
+    let walletName = edgeWallet.name ?? ''
     const receiveAddress = await edgeWallet.getReceiveAddress()
     const enabledTokens = await edgeWallet.getEnabledTokens()
+    const contractAddress = edgeWallet.currencyInfo.metaTokens.find(token => token.currencyCode === currencyCode)?.contractAddress
```

Comment:
This doesn't account for custom tokens, but then again, those don't have images on our system anyhow. This code is fine, and gets the job done.

The `currencyInfo.metaTokens` field is deprecated, though. The information has moved to the `EdgeCurrencyConfig` object, which in this context would be accessed as:

```js
const currencyConfig = this._state.core.account.currencyConfig[edgeWallet.currencyInfo.pluginId]
currencyConfig.builtinTokens // map of tokenId -> EdgeToken
currencyConfig.customTokens // map of tokenId -> EdgeToken
```

I have a task to make it available as `EdgeCurrencyWallet.currencyConfig`, which would be much shorter to type than this crazy-long thing.